src/MLJ.jl

```diff
@@ -5,25 +5,18 @@ function MMI.fit(model::EvoTypes, verbosity::Int, A, y, w = nothing)
     else
         fitresult, cache = init_evotree(model; x_train = A.matrix, y_train = y, w_train = w)
     end
-    grow_evotree!(fitresult, cache)
+    while cache[:info][:nrounds] < model.nrounds
+        grow_evotree!(fitresult, cache, model)
+    end
     report = (features = A.names,)
     return fitresult, cache, report
 end
 
-function okay_to_continue(new, old)
-    new.nrounds - old.nrounds >= 0 &&
-        new.lambda == old.lambda &&
-        new.gamma == old.gamma &&
-        new.max_depth == old.max_depth &&
-        new.min_weight == old.min_weight &&
-        new.rowsample == old.rowsample &&
-        new.colsample == old.colsample &&
-        new.nbins == old.nbins &&
-        new.alpha == old.alpha &&
-        new.device == old.device
+function okay_to_continue(model, fitresult, cache)
+    return model.nrounds - cache[:info][:nrounds] >= 0 &&
+           all(get_types(model) .== get_types(fitresult))
```

Comment:
You sure this is correct? Does `get_types` exclude `:nrounds`, then?

You can also use `MLJModelInterface.is_equal_except(new, old, :nrounds)`.

Reply:
Yes, at least it's aligned with that I intended to do, that is to relax the check so that other training configuration can change throughout training, for example the eta, or some other regularization term. The `get_types` enforce the check on the loss type, the floating point, and the output size.